### PR TITLE
Adder contract

### PR DESCRIPTION
**Pull request by hjorthjort**
Running the adder contract with the Mandos test suite

### CODE REVIEW

**Comment by hjorthjort** [comment]
TODO: More unit tests for host functions

**Comment by hjorthjort** [comment]
TODO: Update to new adder test here: https://github.com/ElrondNetwork/elrond-wasm-rs/tree/master/examples/adder/mandos
Need to build the sources with `./build_wasm.sh`, add that to Makefile, too.

**Comment by hjorthjort** [comment]
This now has tests for all the implemented host functions

File: elrond.md [hunk 1]
```diff
@@ -205,13 +210,312 @@ Parallelized semantics can be achieved by instead using the following rule:
...
+
+#### Call State
+
+```k
+    rule <instrs> hostCall("env", "getNumArguments", [ .ValTypes ] -> [ i32 .ValTypes ]) => i32.const size(ARGS) ... </instrs>
+         <callingArguments> ARGS </callingArguments>
+
+    rule <instrs> hostCall("env", "getArgumentLength", [ i32 .ValTypes ] -> [ i32 .ValTypes ]) => i32.const lengthArg({ARGS[IDX]}:>Argument) ... </instrs>
+         <locals> 0 |-> <i32> IDX </locals>
+         <callingArguments> ARGS </callingArguments>
```
Comment (andrei-marinica) [inline elrond.md:257]: Will crash if IDX >= length of ARGS, not important now, but we should keep in mind

**Review by andrei-marinica — APPROVED** [review]

File: elrond.md (hunk 1 above)
Comment (hjorthjort) [reply elrond.md:257]: You're right, let's guard this with a requires-clause.

**Pull request merged**